crates/tauri-utils/src/config.rs

```diff
@@ -1773,6 +1773,404 @@ pub struct AssetProtocolConfig {
+  fn from(val: HeaderSource) -> Self {
+    match val {
+      HeaderSource::Inline(s) => {
+        let mut out: Vec<String> = Vec::new();
+        let mut separator = ',';
+        if s.contains(';') {
+          separator = ';';
+        }
+        for item in s.split(separator) {
+          out.push(String::from(item));
+        }
+        out
+      }
+      HeaderSource::List(l) => l,
+      HeaderSource::Map(m) => {
+        let mut out: Vec<String> = Vec::new();
+        for (key, value) in m.iter() {
+          let mut item = String::new();
+          item.push_str(key);
+          item.push(' ');
+          item.push_str(value);
+          out.push(item);
+        }
+        out
+      }
+    }
+  }
+}
+
+impl From<HeaderSource> for HashMap<String, String> {
+  fn from(val: HeaderSource) -> Self {
+    match val {
+      HeaderSource::Inline(s) => {
+        let mut out: HashMap<String, String> = HashMap::new();
+        for item in s.split(';') {
+          let index = item.find(' ');
+          if index.is_some() {
+            let (key, value) = item.split_at(index.unwrap());
+            out.insert(String::from(key), String::from(value));
+          }
+        }
+        out
+      }
+      HeaderSource::List(l) => {
+        let mut out: HashMap<String, String> = HashMap::new();
+        for item in l.iter() {
+          let index = item.find(' ');
+          if index.is_some() {
+            let (key, value) = item.split_at(index.unwrap());
+            out.insert(String::from(key), String::from(value));
+          }
+        }
+        out
+      }
+      HeaderSource::Map(m) => m,
+    }
+  }
+}
+
+impl From<String> for HeaderSource {
+  fn from(value: String) -> Self {
+    Self::Inline(value)
+  }
+}
+
+impl From<Vec<String>> for HeaderSource {
+  fn from(value: Vec<String>) -> Self {
+    Self::List(value)
+  }
+}
+
+impl From<HashMap<String, String>> for HeaderSource {
+  fn from(value: HashMap<String, String>) -> Self {
+    Self::Map(value)
+  }
+}
+
+/// A trait which implements on the [`Builder`] of the http create
+///
+/// Must add headers defined in the tauri configuration file to http responses
+pub trait HeaderAddition {
+  /// adds all headers defined on the config file, given the current HeaderConfig
+  fn add_configured_headers(self, headers: Option<&HeaderConfig>) -> Builder;
+}
+
+impl HeaderAddition for Builder {
+  /// Add the headers defined in the tauri configuration file to http responses
+  ///
+  /// this is a utility function, which is used in the same way as the `.header(..)` of the rust http library
+  fn add_configured_headers(mut self, headers: Option<&HeaderConfig>) -> Builder {
+    if let Some(headers) = headers {
+
+        // Add the header Access-Control-Allow-Credentials, if we find a value for it
+        if let Some(value) match &headers.access_control_allow_credentials {
+          self = self.header("Access-Control-Allow-Credentials", value.to_string());
```

Comment:
passing `&str` should be fine here, no need to call `to_string`
```suggestion
          self = self.header("Access-Control-Allow-Credentials", value.as_str());
```

Reply:
The `HeaderSource` value is not always a `String`, so we can't just slice the value with `as_str`. A new String gets created instead, following the rules defined in `enum HeaderSource`
```rust
pub enum HeaderSource {
  /// string version of the header Value
  Inline(String),
  /// list version of the header value. Item are joined by "," for the real header value
  List(Vec<String>),
  /// (Rust struct | Json | JavaScript Object) equivalent of the header value. Items are composed from: key + space + value. Item are then joined by ";" for the real header value
  Map(HashMap<String, String>),
}
```

Reply:
tbh I am not sure why we would support all of these variants, shouldn't be the `Inline` variant enough?

Reply:
Wouldn't the same be true for the Content-Security-Policy (csp)?
I used the csp as a guide to design this option.

Reply:
oh, it is fine to keep then